src/cross_section_measurement/make_control_plots_fromTrees.py

```diff
@@ -11,12 +11,37 @@
 from tools.hist_utilities import prepare_histograms, clean_control_region, get_normalisation_error, get_fitted_normalisation
 from tools.ROOT_utils import get_histograms_from_trees, set_root_defaults
 from tools.latex import setup_matplotlib
+from uncertainties import ufloat
 
 # latex, font, etc
 setup_matplotlib()
 
 title_template = '$%.0f$ pb$^{-1}$ (%d TeV)'
 
+def getPUWeights(histograms_to_draw, histogram_lables) :
+    print 'In get pu weights'
+    # hists = dict(zip(histogram_lables, histograms_to_draw))
+    hists = {}
+    print histogram_lables
+    dataHist = None
+    mcHist = None
+    for label, histogram in zip(histogram_lables, histograms_to_draw):
+        print label
+        if label == 'data':
+            dataHist = histogram.Clone()
+        else :
+            if mcHist == None:
+                mcHist = histogram.Clone()
+            else:
+                mcHist += histogram
+    dataValues = list(dataHist.y())
+    mcValues = list(mcHist.y())
+
+    weights = [ data / mc for data, mc in zip(dataValues, mcValues)]
+    print list(dataHist.x())
+    print list(dataHist.xedges())
+    print weights
```

Comment:
These print out the pu weights for use elsewhere (not used anywhere yet though).